doc/account.md

```diff
@@ -33,31 +36,55 @@ The `passwordSaltLength` is used to adjust the length of the password salt we ge
 
-Initiating an Account
----------------------
 
 The first action for an account is `initiate`. This sets up a few pieces of data needed to properly generate and track an account. We generate a `regId`, `mfaKey`, and `passwordSalt`. An email address must be passed in. These are then passed back to the user to store on their own and will be used to generate their password.
 
 Information will be saved to a file named after the hash of the `regId` in the `registrationDir` [storage] location. This registration file will be deleted upon successful creation of an account file.
 
-Confirming an Account
----------------------
+### Confirming an Account
 
 The next action is to confirm an account. This requires the user to access the URL using the `regId` we gave them. We will then compare it's valid by comparing the hashes then return the `mfaKey` and `passwordSalt` so they can proceed to the next step.
 
-Completing an Account
----------------------
+### Completing an Account
 
 To complete an account, the client will need to pass in a hashed `password`, the `regId` which we gave them previously as well as a `currentMfa` code and `previousMfa` code. This will allow us to find the registration file using the hashed `regId` and looking it up in [storage]. Getting the `mfaKey` off the account we are able to then validate the user using the MFA codes they passed in.
 
 If the account validates, we create the account file from the registration file, add the `password` to it, and generate an `accountId`. The file is then saved under the `accountDir` using the hashed `accountId` as the file name. Only the `accountId` is sent back to the client.
 
+## Logging In
+
+Logging into OpenToken.io is a two step process, involving sending information for the client to create a password hash and OpenToken.io validating.
+
+### Initiating a Login
+
+The client will call into the application using the hashed account id and the `accountId`. This creates and login file containing the `challengeId`, and a `salt` to use for logging in. This file also contains the `accountId`.This file is then sent to [storage].
+
+Information will then be sent to the client for them to create the password for logging in. This contains the `salt` which was just created for the challenge, as well as the `hash` we are expecting them to use once they have created a password hash the same was they did for confirming signup. We also send back the data used when the password was created.
+
+### Completing a Login
+
+Once the password is created on the client side and passed back to OpenToken.io, the original password sent to use is looked up using the login file containing the `accountId`. Using the original `password` and `passwordSalt` from the file, the application then hashes it the same way using the same data the client used and then compares the hashes. If they match, we authorize the [multifactor authentication](hotp.md) token sent in. If that is valid the login is complete.
```

Comment:
Reword the bit about creating a password.  They don't create a password.

They have a password.
They generate a secure hash of the password.
Finally the challenge hash is created.

Maybe those are the terms we should use?  "password", "secure hash" and "challenge hash" in order to help keep this separate in people's minds?  (People includes us too.)